github/github.go

```diff
@@ -174,20 +180,21 @@ func NewClient(httpClient *http.Client) *Client {
 	uploadURL, _ := url.Parse(uploadBaseURL)
 
 	c := &Client{client: httpClient, BaseURL: baseURL, UserAgent: userAgent, UploadURL: uploadURL}
-	c.Activity = &ActivityService{client: c}
-	c.Authorizations = &AuthorizationsService{client: c}
-	c.Gists = &GistsService{client: c}
-	c.Git = &GitService{client: c}
-	c.Gitignores = &GitignoresService{client: c}
-	c.Issues = &IssuesService{client: c}
-	c.Organizations = &OrganizationsService{client: c}
-	c.PullRequests = &PullRequestsService{client: c}
-	c.Repositories = &RepositoriesService{client: c}
-	c.Search = &SearchService{client: c}
-	c.Users = &UsersService{client: c}
-	c.Licenses = &LicensesService{client: c}
-	c.Migrations = &MigrationService{client: c}
-	c.Reactions = &ReactionsService{client: c}
+	c.common.client = c
```

Comment:
Since this optimization is quite small and subtle, I think it'd be nice to leave a small, unobtrusive comment here.

That would reduce the chances of someone accidentally reverting this optimization a year from now, because they may not immediately understand why the code is arranged in this way instead of something simpler.

Something like:

```Go
c.common.client = c // Reuse a single struct instead of allocating one for each service on the heap.
```

(Or something along those lines, phrased in a better way.)

Placing it inline, on the right, is a cue to only read the comment it if you care about editing this section of code.

It's up to you, if you think it's worth it.